shingled_disassembly/conservative_disasm.ml

```diff
@@ -0,0 +1,53 @@
+open Core.Std
+open Or_error
+open Bap.Std
+open Bap_plugins.Std
+module Dis = Disasm_expert.Basic
+
+exception Create_mem of Error.t
+module Lifted_image=Lifted_image_piqi
+
+let lifted_of_bil instruction length offset =
+  { Lifted_image.Lifted_instruction.instruction=Bil_piqi.stmts_to_piqi instruction; length; offset }
+
+let jump global_mem base_addr max_addr lift_insn state mem insn (current_elem, bil_disasm) =
+  (match lift_insn mem insn with
+  | Error e -> ()
+  | Ok bil -> let len = List.length bil in
```

Comment:
### General Note
`length` is O(n), and since all that you want is to know whether it is empty or not, you should use:
```ocaml
match bil with
| [] -> ()
| bil -> ...
```
Or, in this particular case, you can use `if bil = [] then ...`, or even better `if bil <> [] then`  and omit else clause.

### Semantics note

I'm not sure, that you should skip instructions that has an empty bil. This denotes an instruction, that has no visible side effect. Not visible from the BIL perspective. Examples follow: `nop, sync, in, out, cli, clts, hlt, invd` and many more. In short, these are system level instructions an synchronization primitives. They all modeled  with empty bil program, that basically says, that observable state doesn't change/